src/main/java/DilemmaDetector/ScenarioReader.java

```diff
@@ -101,30 +147,28 @@ public Model getModel(int scenarioNumber) {
             });
             vehicles.put(lane, new ArrayList<Vehicle>() {
             });
-        }
-        lanes.put(Model.Side.LEFT, lanes_left);
 
-        for (Vehicle v : scenario.getHas_vehicle()) {
-            String vehicleName = v.getOwlIndividual().getIRI().toString();
-            if (!vehicleName.contains("vehicle_main")) {
-                Lane lane = null;
-                for (Lane l : v.getIs_on_lane()) {
-                    lane = l;
+            for(Vehicle v: lane.getLane_has_vehicle()) {
+                String vehicleName = v.getOwlIndividual().getIRI().toString();
+                if (!vehicleName.equals(mainVehicle.getOwlIndividual().getIRI().toString())) {
+                    v = getVehicleAsSpecificClass(vehicleName);
+                    vehicles.get(lane).add(v);
```

Comment:
Linia 151 do 155 jest dokładnie taka sama jak 117 do 121
Właściwie to chyba nawet 151 - 168 jest takie samo jak 117 - 135

Reply:
nawet lepiej, w 3 miejscach masz 17 linii dokładnie takich samych, różniących się tylko zmienną `lane`